src/cluster_legacy.c

```diff
@@ -2990,7 +2991,51 @@ int clusterProcessPacket(clusterLink *link) {
 
                 if (clusterNodeIsMaster(sender)) {
                     /* Master turned into a slave! Reconfigure the node. */
-                    clusterDelNodeSlots(sender);
+                    if (master && !memcmp(master->shard_id, sender->shard_id, CLUSTER_NAMELEN)) {
+                        /* 'sender' was a primary and was in the same shard as `master`, its new primary */
+                        if (sender->configEpoch > senderConfigEpoch) {
+                            serverLog(LL_WARNING,
```

Comment:
I don't think this is a warning, this seems like a normal state? Warnings are supposed to indicate errors to take action on.
```suggestion
                            serverLog(LL_NOTICE,
```